File: Codenotch.Maui/Models/AgentSession.cs
using System;

namespace Codenotch.Models;

/// <summary>
/// Represents a session of an agent.
/// </summary>
public record AgentSession(
    string Id,
    string Name,
    string Detail,
    AgentSession.SessionState State,
    string? WaitingFor,
    DateTime Since)
{
    public enum SessionState
    {
        Busy,
        Waiting,
        Idle
    }

    public bool IsActive => State == SessionState.Busy || State == SessionState.Waiting;
}

File: lib/sessions/agent_session.dart
enum AgentSessionState {
  busy,
  waiting,
  idle,
}

/// One agent session, whichever tool it belongs to.
class AgentSession {
  final String id;
  final String name;
  final String detail;
  final AgentSessionState state;
  final String? waitingFor;
  final DateTime since;

  const AgentSession({
    required this.id,
    required this.name,
    required this.detail,
    required this.state,
    this.waitingFor,
    required this.since,
  });

  @override
  bool operator ==(Object other) =>
      identical(this, other) ||
      other is AgentSession &&
          runtimeType == other.runtimeType &&
          id == other.id &&
          state == other.state &&
          detail == other.detail &&
          since == other.since;

  @override
  int get hashCode => id.hashCode ^ state.hashCode ^ since.hashCode;
}
